WIP: jsem proposal

sheaf opened the pull request:
PR to this repository for feedback before submitting to the main GHC proposals repository.

adamgundry commented on proposals/0000-jsem.rst:
This could do with more discussion I think, probably in its own subsection. What are the trade-offs between having a single `--make` session using MHU vs using `-jsem`? Do we have reason one would be easier to implement in `cabal` or `stack` or `nix`?

sheaf replied:
I've fleshed it out into a subsection based off today's discussion.

adamgundry commented on proposals/0000-jsem.rst:
There is too much indentation here, so the rendering isn't quite right.

adamgundry commented on proposals/0000-jsem.rst:
Wrong URL? This is a duplicate of the reference above.

adamgundry commented on proposals/0000-jsem.rst:
I would just delete this section. In practice it is rarely used in proposals, and should be dropped IMHO.

adamgundry reviewed (approved):
Looking good. I made a few tiny suggestions.

sheaf committed:
Propose jsem (parallelism semaphores)

This proposes a mechanism for scheduling parallelism across multiple
GHC invocations through use of a system semaphore.

mpickering committed:
updates

mpickering committed:
Some downsides to make jobserver

mpickering committed:
options

sheaf committed:
editing pass, reorganise 'alternatives' section

sheaf committed:
update link to cabal implementation

sheaf committed:
address comments by Simon

sheaf committed:
address Eric's comments

sheaf committed:
Apply suggestion from michaelpj

sheaf committed:
address review comments by Simon & add some benchmark info

sheaf committed:
more info about system semaphores

sheaf committed:
Use RST syntax for multiple home units link

Co-authored-by: Pi Delport <pjdelport@gmail.com>

sheaf committed:
address further comments by Simon

sheaf committed:
clarify 'unit of work'

sheaf committed:
passing semaphore names & coordination

sheaf committed:
final nits
